### PR TITLE
Allow aliases in update statements

### PR DESCRIPTION
**Pull request by dweitzman**
Fix for https://github.com/youtube/vitess/issues/3547

### CODE REVIEW

**Comment by dweitzman** [comment]
Signed the CLA

File: data/test/vtgate/dml_cases.txt [hunk 1]
```diff
@@ -121,6 +121,62 @@
...
+      "Sharded": true
+    },
+    "Query": "update user as user_alias set val = 1 where user_alias.id = 1",
+    "Vindex": "user_index",
+    "Values": [1],
+    "Table": "user"
+  }
+}
+
+# update by primary keyspace id with multi-part where clause
```
Comment (sougou) [inline data/test/vtgate/dml_cases.txt]: The where clause needs to be parenthesized, which is what the new code fixes.

**Comment by dweitzman** [comment]
Added back the parens in the test

**Comment by sougou** [comment]
LGTM

**Pull request merged**